tests/test_ell1h.py

```diff
@@ -12,111 +13,223 @@
-        for pn, p in f.model.get_params_dict("free", "quantity").items():
-            op = getattr(f.model_init, pn)
-            diff = np.abs(p.value - op.value)
-            sigma = diff / op.uncertainty_value
-            # Fit th
-            assert (
-                sigma < 0.7
-            ), "refit %s is %lf sigma different from original value" % (pn, sigma)
-
-    def test_J0613_STIG(self):
-        log = logging.getLogger("TestJ0613.fit_tests_stig")
-        f = ff.GLSFitter(self.toasJ0613, self.modelJ0613_STIG)
-        f.fit_toas()
-        f.model.free_params = ("H3", "STIGMA")
-        for pn, p in f.get_params_dict("free", "quantity").items():
-            op = getattr(f.model_init, pn)
-            diff = np.abs(p.value - op.value)
-            sigma = diff / op.uncertainty_value
-            # Fit th
-            assert (
-                sigma < 0.7
-            ), "refit %s is %lf sigma different from original value" % (pn, sigma)
-
-
-if __name__ == "__main__":
-    pass
+                tol = 1e-3
+            log.debug(
+                "derivative relative diff for %s, %lf"
+                % ("d_delay_d_" + p, np.nanmax(relative_diff).value)
+            )
+            assert np.nanmax(relative_diff) < tol, msg
+        else:
+            continue
+
+
+def test_J0613_H4(toasJ0613, modelJ0613):
+    log = logging.getLogger("TestJ0613.fit_tests")
+    f = ff.GLSFitter(toasJ0613, modelJ0613)
+    f.fit_toas()
+    f.model.free_params = ("H3", "H4")
+    for pn, p in f.model.get_params_dict("free", "quantity").items():
+        op = getattr(f.model_init, pn)
+        diff = np.abs(p.value - op.value)
+        sigma = diff / op.uncertainty_value
+        # Fit th
+        assert sigma < 0.7, "refit %s is %lf sigma different from original value" % (
+            pn,
+            sigma,
+        )
+
+
+def test_J0613_STIG(toasJ0613, modelJ0613_STIG):
+    log = logging.getLogger("TestJ0613.fit_tests_stig")
+    f = ff.GLSFitter(toasJ0613, modelJ0613_STIG)
+    f.fit_toas()
+    f.model.free_params = ("H3", "STIGMA")
+    for pn, p in f.get_params_dict("free", "quantity").items():
+        op = getattr(f.model_init, pn)
+        diff = np.abs(p.value - op.value)
+        sigma = diff / op.uncertainty_value
+        # Fit th
+        assert sigma < 0.7, "refit %s is %lf sigma different from original value" % (
+            pn,
+            sigma,
+        )
+
+
+def test_no_H3_H4(toasJ0613):
+    """ Test no H3 and H4 in model.
+    """
+    no_H3_H4 = simple_par.replace("H4 2.0262048E-7  1       1.1276173E-7", "")
+    no_H3_H4 = no_H3_H4.replace("H3 2.7507208E-7  1       1.5114416E-7", "")
+    print(no_H3_H4)
+    no_H3_H4_model = model.get_model(StringIO(no_H3_H4))
+    assert no_H3_H4_model.H3.value == None
+    assert no_H3_H4_model.H4.value == None
+    test_toas = toasJ0613[::20]
+    f = ff.WLSFitter(test_toas, no_H3_H4_model)
+    f.fit_toas()
+
+
+def test_H3_H4_pairs(toasJ0613):
```

Comment:
Er, this test only seems to test one model? Which of H3, H4 are zero here? Nicest if the function name makes that clear (nobody ever needs to call these functions so long names are not a problem).